torch/csrc/jit/codegen/cuda/root_domain_map.cpp

```diff
@@ -1106,6 +1124,28 @@ void ComputeAtRootDomainMapBuilder::handle(GatherOp* op) {
   }
 }
 
+void ComputeAtRootDomainMapBuilder::handle(TorchGatherOp* op) {
```

Comment:
Why did you need this? I does make sense, and I think we should need something similar for `select` and `index_select`, but I though we're still fine without this.

Reply:
When `index` tensor has a broadcast domain, and the `torch_gather` will look like `T[iS2, bS3] = torch_gather(input = T[iS0, iS1], dim = 0, index = T[iS2, bS3])`. If we use `mapPointwiseOrReductionOp` to handle `torch_gather`, the `iS1` and `bS3` will miss match.

Reply:
And the miss match will lead to an `Assert` in [this line](https://github.com/csarofeen/pytorch/blob/74b773ec2acd503082165550db0d14beecbee0b1/torch/csrc/jit/codegen/cuda/root_domain_map.cpp#L860)

Reply:
OK, that makes sense.